dev/tests/js/jasmine/tests/app/code/Magento/AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.test.js

```diff
@@ -0,0 +1,147 @@
+                var type = typeof overlayObj.initObservable;
+
+                expect(type).toEqual('function');
+            });
+
+            it('licensed will be defined', function () {
+                expect(overlayObj.licensed).toBeDefined();
+            });
+        });
+
+        describe('"updateLicensed" method', function () {
+            it('Check for defined', function () {
+                expect(overlayObj.hasOwnProperty('updateLicensed')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.updateLicensed;
+                expect(type).toEqual('function');
+            });
+
+            it("Check Ajax Success request", function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: true
+                        };
+                    }
+                }, ids = [1,2,3];
+                spyOn(overlayObj, 'login').and.callFake(function(e) {
+                    return login;
+                });
+                spyOn(overlayObj, 'getIds').and.callFake(function(e) {
+                    return ids;
+                });
+                spyOn($, 'ajax').and.callFake(function(e) {
+                    return $.Deferred().resolve({
+                        'response': 'success'
+                    }).promise();
+                });
+                overlayObj.updateLicensed();
+                expect(overlayObj.updateLicensed()).toBeUndefined();
+                expect($.ajax).toHaveBeenCalled();
+                expect(overlayObj.getIds).toHaveBeenCalled();
+            });
+
+            it('Check Ajax failure request', function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: false
+                        };
+                    }
+                }, ids = [1,2,3];
+                spyOn(overlayObj, 'login').and.callFake(function(e) {
+                    return login;
+                });
+                spyOn(overlayObj, 'getIds').and.callFake(function(e) {
+                    return ids;
+                });
+                spyOn($, 'ajax');
+                overlayObj.updateLicensed();
+                expect(overlayObj.updateLicensed()).toBeUndefined();
+                expect($.ajax).not.toHaveBeenCalled();
+                expect(overlayObj.getIds).not.toHaveBeenCalled();
+            });
+        });
+
+        describe('"getIds" method', function () {
+
+            it('Check for defined ', function () {
+                expect(overlayObj.hasOwnProperty('getIds')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.getIds;
+                expect(type).toEqual('function');
+            });
+
+        });
+
+        describe('"getStyles" method', function () {
```

Comment:
Same as `getIds`, this set of tests do not exercise `getStyles` logic. It seems to me like the logic of `getStyles` may fail in certain cases. The logic that calculates the `height` variable in this method, for example, is mixing string and numbers in JavaScript and using arithmetic operators on mixed string and number types can yield unexpected results 😄. I think it would be a good idea to write logic tests for this method as I think we may catch a bug here 😄